### PR TITLE
Run with class labels.

### PR DESCRIPTION
**Pull request by MylesBartlett**
Adapted training/testing scripts to accommodate class information.
Adapted models to models to allow use of class information.

### CODE REVIEW

File: optimisation/testing.py [hunk 1]
```diff
@@ -34,15 +34,17 @@ def test(args, sample_transform):
...
     with torch.no_grad():
         for img_no, sample in enumerate(tqdm(test_loader)):
             noisy = sample['noisy']
-            noisy = noisy.cuda() if args.cuda else noisy
             iso = sample['iso']
+            class_labels = sample['class']
+
+            noisy = noisy.cuda() if args.cuda else noisy
             iso = iso.cuda() if args.cuda else iso
+            class_labels = class_labels.cuda() if args.cuda else class_labels
```
Comment (Stoffle) [inline optimisation/testing.py:45]: Have you tested this? Seems like doing this to a string will throw an exception, and that's what `sample['class']` is for the `sample_transform` being used

Comment (MylesBartlett) [reply optimisation/testing.py:45]: It certainly won't work if it's a string; I'll change it so that class labels are loaded in as a 1-dimensional tensor containing the corresponding class values. That seems like the cleanest approach.

File: utils/transform_data.py [hunk 2]
```diff
@@ -27,39 +27,35 @@ def main(patches, patch_size, old_path=None, new_path=None):
...
         for i in range(patches):
-            noisy = transform(pair['noisy'])
+            noisy = transform(sample['noisy'])
             noisy.save(noisy_path / f"{i}.png")
-        dataset_info_writer(noisy_path, clean_path, pair['iso'])
-    with open(Path(transformed_path).resolve() / "info.csv", 'w') as csv_file:
-        writer = csv.writer(csv_file)
-        writer.writerows(iso_data)
+        dataset_info_writer(noisy_path, clean_path, sample['iso'], sample['class'])
+    data.info_df.to_csv(str(transformed_path / "Training_Data.csv"), index=False)
```
Comment (tmke8) [inline utils/transform_data.py:65]: Does this really give the same result?

Comment (Stoffle) [reply utils/transform_data.py:65]: no, but it's not supposed to - this is a change from one of my earlier commits (adding support for image classes). It just makes a copy of the original csv inside the transformed data folder so there's not another path to specify
